monkey/infection_monkey/island_api_client/http_client.py

```diff
@@ -122,17 +125,15 @@ def _send_request(
         *args,
         **kwargs,
     ) -> requests.Response:
-        if self._api_url is None:
-            raise RuntimeError(
-                "HTTP client is not connected to the Island server,"
-                "establish a connection with 'connect()' before "
-                "attempting to send any requests"
-            )
-        url = f"{self._api_url}/{endpoint}".strip("/")
+        if self._server_url is None:
+            raise RuntimeError("HTTP client does not have a server URL set")
+        url = f"{self._server_url.strip('/')}/{endpoint.strip('/')}".strip("/")
```

Comment:
It's to allow sending requests to the server url without endpoints (for example if the server url is `https://1.1.1.1:4000/api` we will be able to send `https://1.1.1.1:4000/api?action=reset` by passing params and `endpoint=""`)

Reply:
1. You mean if the endpoint is `None` or `""`? Won't calling `.strip()` on `None` raise an error? Let's default it to `""`.
2. 
```suggestion
        url = "/".join([self._server_url.strip('/'), endpoint.strip('/')]
```

Reply:
1. Endpoint can't be `None`. Setting the default to "" is a good idea.
2. this will result in `https://1.1.1.1:4000/api/?action=reset` and I don't like join for two elements